build.js

```diff
@@ -124,32 +59,69 @@ const injectFirstHitJs = async (metafile, revision) => {
  */
 const injectHtmlPages = async (metafile, revision) => {
   const htmlFilePaths = await fs.readdir(path.resolve('dist'), { withFileTypes: true })
-    .then(files => files.filter(file => file.isFile() && file.name.startsWith('ipfs-sw-') && file.name.endsWith('.html') && !file.name.includes('first-hit')))
+    .then(files => files.filter(file => file.isFile() && file.name.endsWith('.html')))
     .then(files => files.map(file => path.resolve('dist', file.name)))
 
+  // htmlFilePaths.push(path.resolve('dist/index.html'))
+
   for (const htmlFilePath of htmlFilePaths) {
-    // find the -index-*.css file in the metafile results
-    const cssFile = Object.keys(metafile.outputs).find(file => file.endsWith('.css') && file.includes('ipfs-sw-index'))
+    const baseName = path.basename(htmlFilePath, '.html')
+    let jsFile = Object.keys(metafile.outputs).filter(file => file.endsWith('.js') && (file.includes(baseName) || file.includes('index')))
+    if (jsFile.length > 1) {
+      // override injection of index.js to the basename file
+      jsFile = jsFile.find(file => file.includes(baseName))
+    } else {
+      jsFile = jsFile[0]
+    }
 
-    if (!cssFile) {
-      console.error('Could not find the ipfs-sw-index CSS file in the metafile.')
-      return
+    let cssFile = Object.keys(metafile.outputs).filter(file => file.endsWith('.css') && (file.includes(baseName) || file.includes('index')))
+    if (cssFile.length > 1) {
+      // override injection of index.css to the basename file
+      cssFile = cssFile.find(file => file.includes(baseName))
+    } else {
+      cssFile = cssFile[0]
     }
```

Comment:
we search for both a `ipfs-sw-<name>.*` override .js or .css file, and then take that override if it exists.